platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +97,155 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
     @Override
     public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
+        hasBeenInitialized = true;
+
+        // now that the DelegateServletFilter has been initialized with a filterConfig, initialize all SecurityFilters that have not yet been initialized using that filterConfig
+        for (ServiceReference<SecurityFilter> securityFilterServiceReference : notYetInitializedSecurityFilterServiceReferences) {
+            initAndAddToFilterList(securityFilterServiceReference, filterConfig);
+            notYetInitializedSecurityFilterServiceReferences.remove(securityFilterServiceReference);
+        }
+    }
+
+    public void addSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference)
+            throws ServletException {
+        if (notYetInitializedSecurityFilterServiceReferences.contains(securityFilterServiceReference)
+                || sortedAndInitializedSecurityFilterServiceReferences.contains(
+                securityFilterServiceReference)) {
+            LOGGER.warn("SecurityFilter has already been added. Not adding again.");
+        } else {
+            // only initialize SecurityFilters after the DelegateServletFilter has been initialized with a filterConfig
+            if (hasBeenInitialized) {
+                initAndAddToFilterList(securityFilterServiceReference, filterConfig);
+            } else {
+                notYetInitializedSecurityFilterServiceReferences.add(securityFilterServiceReference);
+            }
+        }
+    }
+
+    private void initAndAddToFilterList(
+            ServiceReference<SecurityFilter> securityFilterServiceReference,
+            FilterConfig filterConfig) throws ServletException {
+        final ServletContext servletContext;
+
+        if (filterConfig == null) {
+            LOGGER.error(
+                    "DelegateServletFilter was initialized with a null filterConfig. Initializing SecurityFilter {} with null ServletContext",
+                    ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            servletContext = null;
+        } else {
+            servletContext = this.filterConfig.getServletContext();
+        }
+
+        bundleContext.getService(securityFilterServiceReference)
```

Comment:
After sleeping on it I see what is going on here. It still feels a little weird to call an init method on a service.